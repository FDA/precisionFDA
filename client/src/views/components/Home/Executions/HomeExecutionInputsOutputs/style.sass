.home-execution-inp-out
  &__grid
    display: grid;
    grid-template-columns: auto auto
  
  &__table
    padding: 10px 15px
    width: 50%

    &_title
      color: #8198BC
      text-transform: uppercase
      font-weight: 400
      font-size: 14px
      margin-bottom: 10px
      padding-left: 8px

    &_row
      display: flex
      padding: 8px
      border-top: 1px solid #ddd

      &_even
        background: #ebf3fb

    &_type
      color: #8198BC
      font-family: "PT Mono", Menlo, Monaco, Consolas, "Courier New", monospace

  &__container
    background-color: #F4F8FD
    border: 1px solid #ddd
    border-bottom-right-radius: 3px
    border-bottom-left-radius: 3px
    display: flex