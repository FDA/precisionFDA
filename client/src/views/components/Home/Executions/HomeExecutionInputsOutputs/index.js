import React, { Fragment } from 'react'
import classNames from 'classnames'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import './style.sass'


const Table = ({ title, config }) => {
  const list = config.map((e, i) => {
    const classes = classNames({
      'home-execution-inp-out__table_row': true,
      'home-execution-inp-out__table_row_even': !(i % 2),
    })

    return (
      <Fragment key={i}>
        <div className={classNames(classes, 'home-execution-inp-out__table_type')}>{e.label}</div>
        {e.link ?
          <Link to={e.link} className={classes}>{String(e.value)}</Link> :
          <div className={classes}>{String(e.value)}</div>
        }
      </Fragment>
    )
  })

  return (
    <div className='home-execution-inp-out__table'>
      <div className='home-execution-inp-out__table_title'>{title}</div>
      <div className='home-execution-inp-out__grid'>
        {list}
      </div>
    </div>
  )
}

const HomeExecutionInputsOutputs = ({ runInputData, runOutputData }) => {
  const getConfig = (config) => {
    return config.map((e) => {
      let link = ''
      let value = e.value

      if (e.class === 'file') {
        value = e.file_name
        link = `/home/files/${e.file_uid}`
      }

      return {
        label: e.label || e.name,
        link,
        value,
      }
    })
  }

  const inputConfig = getConfig(runInputData)
  const outputConfig = getConfig(runOutputData)

  return (
    <div className='home-execution-inp-out__container'>
      {<Table title='inputs' config={inputConfig} />}
      {<Table title='outputs' config={outputConfig} />}
    </div>
  )
}

HomeExecutionInputsOutputs.propTypes = {
  runDataUpdates: PropTypes.object,
  runInputData: PropTypes.array,
  runOutputData: PropTypes.array,
}

HomeExecutionInputsOutputs.defaultProps = {
  runDataUpdates: {},
  runInputData: [],
  runOutputData: [],
}

Table.propTypes = {
  config: PropTypes.array,
  title: PropTypes.string,
}

export default HomeExecutionInputsOutputs
