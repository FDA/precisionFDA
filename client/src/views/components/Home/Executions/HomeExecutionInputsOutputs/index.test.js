import React from 'react'
import { shallow } from 'enzyme'

import HomeExecutionInputsOutputs from '.'


describe('HomeExecutionInputsOutputs test', () => {
  it('should render', () => {
    const component = shallow(<HomeExecutionInputsOutputs />)

    expect(component).toMatchSnapshot()
  })
})
