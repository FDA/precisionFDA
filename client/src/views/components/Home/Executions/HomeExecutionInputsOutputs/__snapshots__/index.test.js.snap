// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeExecutionInputsOutputs test should render 1`] = `
<div
  className="home-execution-inp-out__container"
>
  <Table
    config={Array []}
    title="inputs"
  />
  <Table
    config={Array []}
    title="outputs"
  />
</div>
`;
